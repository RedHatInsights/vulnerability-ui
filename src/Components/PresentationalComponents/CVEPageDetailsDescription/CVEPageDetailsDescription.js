import { List, ListItem, Stack, StackItem, Text, TextContent, TextVariants } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import './CVEPageDetailsDescription.scss';

class CVEPageDetailsDescription extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const listOfLinks = this.props.links.map((item, index) => <ListItem key={index}>{item}</ListItem>);

        return (
            <React.Fragment>
                <Stack gutter="md">
                    <StackItem>
                        <TextContent>
                            <Text component={TextVariants.p}>{this.props.description}</Text>
                        </TextContent>
                    </StackItem>
                    {this.props.links.length > 0 ? (
                        <StackItem>
                            <List>
                                {listOfLinks}
                            </List>
                        </StackItem>
                    ) : ('')}
                </Stack>
            </React.Fragment>
        );
    }
}

CVEPageDetailsDescription.defaultProps = {
    description: '',
    links: []
};

CVEPageDetailsDescription.propTypes = {
    description: propTypes.string,
    links: propTypes.array
};

export default CVEPageDetailsDescription;
