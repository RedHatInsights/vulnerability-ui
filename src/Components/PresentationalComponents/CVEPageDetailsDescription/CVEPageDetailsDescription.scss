@import "../../../App";

.splitter {
    margin-bottom: var(--pf-global--spacer--sm) !important;
    margin-top: var(--pf-global--spacer--sm) !important;
}
.status-breakdown-table {
    tr {
        th {
            font-weight: normal;
            padding-left: 0px;
            padding-right: var(--pf-global--spacer--md);
            padding-bottom: var(--pf-global--spacer--sm);
            &:last-child {
                text-align: right;
                min-width: 8em;
            }
        }
        td {
            padding-left: 0px;
            padding-right: var(--pf-global--spacer--md);
            white-space: nowrap;
            padding-top: var(--pf-global--spacer--sm);
            padding-bottom: var(--pf-global--spacer--sm);
            &:last-child {
                text-align: right;
            }
        }
    }
}
