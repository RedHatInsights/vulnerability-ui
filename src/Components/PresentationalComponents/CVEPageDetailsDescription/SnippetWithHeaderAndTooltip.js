import { Popover, Stack, StackItem } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import Label from '../Snippets/Label';
import './CVEPageDetailsDescription.scss';

const SnippetWithHeaderAndPopover = props => {
    const { title, value, content } = props;

    return (
        <Popover id="popover" bodyContent={content} headerContent={''} aria-label="Business risk popover" position="right">
            <Stack style={{ display: 'inline-block', cursor: 'pointer' }}>
                <StackItem>
                    <Label>{title}</Label>
                </StackItem>
                <StackItem>
                    <a>{value}</a>
                </StackItem>
            </Stack>
        </Popover>
    );
};

SnippetWithHeaderAndPopover.propTypes = {
    title: propTypes.any,
    value: propTypes.any,
    content: propTypes.any
};
export default SnippetWithHeaderAndPopover;
