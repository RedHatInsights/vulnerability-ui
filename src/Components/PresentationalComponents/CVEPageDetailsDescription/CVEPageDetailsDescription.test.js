import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CVEPageDetailsDescription from './CVEPageDetailsDescription';

describe('CVEPageDetailsDescription component', () => {
    it('should render without props', () => {
        const wrapper = shallow(<CVEPageDetailsDescription />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with description but no links', () => {
        const wrapper = shallow(
            <CVEPageDetailsDescription
                description={
                    'It was found that ghostscript could leak sensitive operators on the operand stack when a pseudo-operator'
                    + ' pushes a subroutine. A specially crafted PostScript file could use this flaw to escape the -dSAFER'
                    + ' protection in order to, for example, have access to the file system and execute commands.'
                }
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
