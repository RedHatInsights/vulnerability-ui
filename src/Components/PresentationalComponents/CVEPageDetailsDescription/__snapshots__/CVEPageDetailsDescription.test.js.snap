// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVEPageDetailsDescription component should render with description but no links 1`] = `
<Fragment>
  <Stack
    className=""
    component="div"
    gutter="md"
  >
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <Text
          className=""
          component="p"
        >
          It was found that ghostscript could leak sensitive operators on the operand stack when a pseudo-operator pushes a subroutine. A specially crafted PostScript file could use this flaw to escape the -dSAFER protection in order to, for example, have access to the file system and execute commands.
        </Text>
      </TextContent>
    </StackItem>
  </Stack>
</Fragment>
`;

exports[`CVEPageDetailsDescription component should render without props 1`] = `
<Fragment>
  <Stack
    className=""
    component="div"
    gutter="md"
  >
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <Text
          className=""
          component="p"
        />
      </TextContent>
    </StackItem>
  </Stack>
</Fragment>
`;
